### DISCUSSION TITLE
Is there a way to use the AIRFLOW_CTX_TRY_NUMBER variable in a task?

### DISCUSSION THREAD

**Discussion by waardd** [apache/airflow#29420] [Q&A] [answered]
I'm trying to get the AIRFLOW_CTX_TRY_NUMBER variable into my script but it does not work.

ive tried:
retrycount = Variable.get('AIRFLOW_CTX_TRY_NUMBER')
print ("Retrycount: "+str(retrycount))

But it fails.

So how do i get to the exported varibale AIRFLOW_CTX_TRY_NUMBER

[2023-02-08, 09:31:11 CET] {taskinstance.py:1581} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=fm.nl.its.rf.abusemca@rabobank.nl
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=Sophis
AIRFLOW_CTX_TASK_ID=copyfile
AIRFLOW_CTX_EXECUTION_DATE=2023-02-08T08:30:57.012422+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-02-08T08:30:57.012422+00:00

**Comment by potiuk** [comment, marked as answer]
You are mixing Airflow Variables and os environment variables. Those are two different types of variables.

**Reply by potiuk** [reply]
https://stackoverflow.com/questions/4906977/how-can-i-access-environment-variables-in-python

**Reply by waardd** [reply]
Aha, i see now.
I tried this and it showed all the variables i need.

for i, j in os.environ.items():
        print(i, j)

**Reply by MichaelNaor** [reply]
Hi @waardd , did you actually get the AIRFLOW_CTX_TRY_NUMBER when running 
for i, j in os.environ.items():
print(i, j)?
I am not getting this variable in the list.
